ui/phase_manager.py

```
@@ -0,0 +1,539 @@
+                - percent (float): Overall completion percentage based on weights, rounded to 1 decimal (0-100).
+                - weighted_completed (float): Sum of completed phase weights, rounded to 1 decimal.
+                - weighted_total (float): Sum of all phase weights.
+        """
+        """
+        Compute the session's phase completion progress and weighted progress values.
+        
+        Returns:
+            progress (dict): Progress snapshot containing:
+                - completed (int): Number of phases marked completed.
+                - total (int): Total number of phases in the session.
+                - percent (float): Overall completion percentage based on weights, rounded to 1 decimal (0–100).
```

Ruff (0.15.9)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)